surveys/micro/2025-06-compiler-performance.md

```diff
@@ -0,0 +1,269 @@
+
+Type: free form (optional)
+
+### How long do you need to wait for the compiler to rebuild your code after making a change?
+
+Please select the longest duration range amongst the projects that you regularly work on.
+
+This question focuses on local development workflows, not CI or remote builds. If you do not compile your Rust code locally, please skip this question.
+
+Type: select one (optional)
+
+- Less than a second
+- Between 1 and 5 seconds
+- Between 5 and 10 seconds
+- Between 10 and 30 seconds
+- Between 30 seconds and 1 minute
+- Between 1 and 5 minutes
+- More than 5 minutes
+
+### Which of the following problems do you most struggle with?
+
+Type: matrix (optional)
+
+Commands:
+
+- Waiting for a rebuild after making a small change
+- Waiting for CI workflows that build Rust code
+- Waiting for an IDE to show me inline error/warning annotations
+- *cargo* and my IDE blocking each other
+- *cargo check* and *cargo build* not sharing compilation cache
+- *cargo check* and *cargo clippy* not sharing compilation cache
+- *cargo* rebuilding everything from scratch and I do not understand why
+
+Options:
+
+- Big problem for me
+- Could be improved, but does not limit me
+- Not an issue for me at all
+
+### How do you primarily examine compilation errors in your code?
+
+Type: select all that apply (optional)
+
+- Using code editor (e.g. inline annotations)
+- Using terminal commands (e.g. *cargo check*)
+- Other (open response)
+
+### If you use Cargo, how often do you use the following commands after each change to Rust code?
+
+Type: matrix (optional)
+
+Commands:
+
+- cargo check
+- cargo clippy
+- cargo test
+- cargo run / cargo build
+
+Frequency:
+
+- After every code change
+- Only after I have resolved other issues
+- Only after I am done (e.g. before committing code)
+- Never
+
+## Workarounds
+
+### Have you used any of the following mechanisms to improve compilation performance?
+
+Please select only mechanisms/workarounds that you have used at least once **primarily** in order to improve compilation performance, not for other reasons.
+
+Type: select all that apply (optional)
+
+- Disable (or reduce) debuginfo (e.g. set *debug = 0* in Cargo.toml)
+- Parallel compiler frontend (pass *-Zthreads=<N>* to the compiler)
+- Cranelift codegen backend (e.g. set *codegen-backend = "cranelift"* in Cargo.toml)
+- Alternative linker (e.g. lld/mold/wild)
+- Caching compiler wrapper (e.g. sccache)
+- Share *target* directory amongst multiple projects (e.g. with CARGO_TARGET_DIR)
+- Split crates into smaller crates
+- Reduce the amount of dependencies
+- Disable default Cargo features of dependencies
+- Create a Cargo feature to opt into building certain dependencies (or their features)
+- Reduce usage of procedural macros
+- Reduce usage of generic code (e.g. by converting it to *dyn Trait* instead)
+- Merge integration tests into a single binary
+- Something else (open response)
```

Comment:
It'd be nice to sort this section by a rough likelihood of an action. I bet most people are familiar with splitting crates up, reducing the amount of dependencies or tweaking the feature set - less so with merging integration tests, sharing target directories or tweaking code structure to make it compile faster.

Reply:
Interesting, I guess you're thinking of avoiding fatigue, as people might be annoyed to read all of them and you think we should offer the most popular options first? From a statistical/bias standpoint, I would actually like to configure these answers to be randomized, to avoid bias by presenting selected choices first :sweat_smile: (we can configure the randomization in SurveyHero, we did it for a bunch of options in the annual survey).

Reply:
>  I guess you're thinking of avoiding fatigue

Yes, exactly. I'm not willing to die on that hill though and I see why you'd want them randomized - I just worry that there are quite a few options and some of the respondents won't know what all of these options even mean.

Reply:
I will actually turns this into a "matrix" answer, to also let us know if these approaches helped users or not. And I will clarify that all answers (and most questions) are optional.